File: 1359-count-all-valid-pickup-and-delivery-options/1359-count-all-valid-pickup-and-delivery-options.cpp
class Solution {
public:
    int countOrders(int n) {
        long long res = 1,mod = 1e9+7;
        for(int i=1;i<=n;i++)
            res = (res*i)%mod;
        for(int i=1;i<2*n;i+=2)
            res = (res*i)%mod;
        return res%mod;
    }
};

File: 1359-count-all-valid-pickup-and-delivery-options/1359-count-all-valid-pickup-and-delivery-options.py
class Solution:
    def countOrders(self, n: int) -> int:
        mod = 10**9 + 7
        pickup_permutation = math.factorial(n)%mod
        delievery_permutation = reduce(mul,range(1,2*n,2),1)%mod
        return pickup_permutation*delievery_permutation%mod